### PR TITLE
parameterize string for generating valid URLs

### PR DESCRIPTION
**Pull request by yagarea**
Hello,
I managed to solve it without need of whole ruby on rails frame work. I find out it is possible to import just specific classes.
This does everything as it should and code is very clean.

### CODE REVIEW

**Comment by yagarea** [comment]
This pull request solves #1 issue

**Comment by yagarea** [comment]
If there is any reason not to accept this pr feel free to point it out and I can try to fix it.

**Comment by NiklasEi** [comment]
I just need some time to set up ruby on my machine to test it. Code looks fine in my opinion, thank you :)

File: lib/jekyll_custom_permalink/custom_permalink.rb [hunk 1]
```diff
@@ -17,7 +19,7 @@ def generate(site)
             raise CustomPermalinkSetupError, "The custom placeholders need to be an array! Check the settings of your '#{doc.collection.label}' collection."
           end
           def doc.url_template
-            @custom_url_template ||= collection.metadata.fetch("custom_permalink_placeholders").inject(collection.url_template){|o,m| o.sub ":" + m, data[m].to_s}
+            @custom_url_template ||= collection.metadata.fetch("custom_permalink_placeholders").inject((collection.url_template){|o,m| o.sub ":" + m, data[m].to_s.parameterize}
```
Comment (NiklasEi) [inline lib/jekyll_custom_permalink/custom_permalink.rb]:
```suggestion
            @custom_url_template ||= collection.metadata.fetch("custom_permalink_placeholders").inject(collection.url_template){|o,m| o.sub ":" + m, data[m].to_s.parameterize}
```

**Review by NiklasEi — COMMENTED** [review]
I got my setup working and ran into a few issues with this. I can't seem to get the dependency to `active_support` working. Shouldn't we add the dependency in the gemspec?
But even after doing this my Jekyll site didn't build and complained about `active_support` being missing.
It would also be nice if you could add something in the changelog and mention in the readme that all values get parameterized.

**Comment by yagarea** [comment]
I was experimenting with `active_support` out of this plugin in some testing scripts. For that I installed it using `gem` globally on my system. Currently I am using the modified version on my site. Unfortunately this is not ideal solution. I added `activesupport` into gemspec file and now it  should be installed when you install this plugin as well.

**Comment by yagarea** [comment]
Thank you for merge

**Comment by NiklasEi** [comment]
Only took 3 years, sorry about that :D
Do you still use the plugin?

**Comment by yagarea** [comment]
> Only took 3 years, sorry about that :D Do you still use the plugin?

Yes, I am using the patched version

**Pull request merged**